manage.py

```diff
@@ -0,0 +1,179 @@
+                    '-t', self.get_full_name(name),
+                ]
+
+        yield command[ROOT_PATH]
+
+        if args.push:
+            yield from self['push'].prepare(args)
+
+        if args.deploy:
+            yield self['deploy'].prepare(args)
+
+    @localmethod('-l', '--login', action='store_true', help="log in to AWS ECR")
+    def push(self, args):
+        """push already-built image to registry"""
+        if args.login:
+            login_command = self.local['aws'][
+                'ecr',
+                'get-login',
+                '--no-include-email',
+                '--region', 'us-west-2',
+            ]
+            if args.show_commands or not args.execute_commands:
+                print('>', login_command)
+            if args.execute_commands:
+                full_command = login_command()
+                (executable, *arguments) = full_command.split()
+                assert executable == 'docker'
+                yield self.local[executable][arguments]
+
+        yield self.local['docker'][
+            'push',
+            self.get_full_name(args.name),
+        ]
+
+    @localmethod
+    def deploy(self, args):
+        """deploy an image container"""
+        # FIXME
+        raise NotImplementedError
+
+        command = self.local['eb']['deploy']
+
+        # specify environment
+        if args.target == 'production':
+            command = command['appy-reviews-pro']
+        else:
+            command = command['appy-reviews-dev']
+
+        if args.label:
+            return command['-l', args.label[0]]
+        return command
+
+
+@Marketplace.register
+class Develop(Local):
+
+    DEFAULT_NAMETAG = 'marketplace_web'
+
+    @classmethod
+    def run(cls, *args, name=None, **kwargs):
+        docker = cls.local['docker']
+        return docker[
+            'run',
+            '-p', '8000:8000',
+            '-v', f'{SRC_PATH}:/app',
+            '--env-file', str(ROOT_PATH / '.env'),
+        ].bound_command(
+            *args
+        ).bound_command(
+            *(
+                f'-e{key}' if value is None else f'-e{key}={value}'
+                for (key, value) in kwargs.items()
+            )
+        )[name or cls.DEFAULT_NAMETAG]
+
+    def __init__(self, parser):
+        super().__init__(parser)
+
+        parser.add_argument(
+            '-n', '--name',
+            default=self.DEFAULT_NAMETAG,
+            help=f'Image name/tag (default: {self.DEFAULT_NAMETAG})',
+        )
+        parser.add_argument(
+            '-b', '--build',
+            action='store_true',
+            help="(re-)build image before container creation",
+        )
+
+    def prepare(self, args):
+        if args.build:
+            yield self.local['docker'][
+                'build',
+                '--build-arg', 'TARGET=development',
+                '-t', args.name,
+                ROOT_PATH,
+            ]
+
+        yield self.run(
+            '-d',
+            '--name', args.name,
+        )
```

Comment:
Some similarities here, but basically a different thing – build the image with development requirements, and run it locally, with the source code mounted into the container, (such that you can make changes to the code in the project directory and see them reflected in the container).